### PR TITLE
ARC 1640 - Server URL screen updates

### PR DESCRIPTION
**Pull request by krazziekay**
**What's in this PR?**
- Time out added for `JiraConnectEnterprisePost`
- Error code passed for default error messages.
- Button click spamming blocked.
- Error validation check done on `keyUp` event
- Question icon removed from manual-app-creation page.

**Why**
- Test fixes 

**How has this been tested?**  
- Local environment

### CODE REVIEW

File: src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts [hunk 1]
```diff
@@ -68,6 +70,15 @@ export const JiraConnectEnterprisePost = async (
 		}
 
 		req.log.debug(`No existing GitHub apps found for url: ${gheServerURL}. Making request to provided url.`);
+
+		// Setting time out check
+		res.setTimeout(TIMEOUT_PERIOD, () => {
```
Comment (bgvozdev) [inline src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts:75]: Would it make sense to also use axios' timeout? 
image

Comment (krazziekay) [reply src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts]: Replaced this by using axios's timeout...

File: src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ import { metricError } from "config/metric-names";
 import { sendAnalytics } from "utils/analytics-client";
 import { AnalyticsEventTypes, AnalyticsTrackEventsEnum } from "interfaces/common";
 
+const TIMEOUT_PERIOD = 30 * 1000;
```
Comment (bgvozdev) [inline src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts:10]: nit: could've been `TIMEOUT_PERIOD_MSEC` for clarity

**Review by bgvozdev — APPROVED** [review]

File: src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts [hunk 3]
```diff
@@ -33,9 +35,9 @@ export const gheServerUrlErrors: GheServerUrlErrors = {
 		errorCode: "GHE_SERVER_BAD_GATEWAY",
 		message: "Bad gateway"
 	},
-	default: {
-		errorCode: "GHE_ERROR_DEFAULT",
-		message: "Something went wrong"
+	ECONNABORTED: {
```
Comment (rachellerathbone) [inline src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts:38]: Do we want to get rid of the default? What will be displayed when an user hits an error we haven't mapped? I think we should leave the default so we still catch unknown errors, and continue to update the error mapping as more errors surface.

**Review by rachellerathbone — CHANGES_REQUESTED** [review]

File: static/js/jira-server-url.js [hunk 4]
```diff
@@ -40,24 +49,35 @@ const requestFailed = () => {
...
+			};
+		case "GHE_SERVER_BAD_GATEWAY":
+			return {
+				title: "Bad Gateway",
+				message: "We weren't able to complete your request. Please try again."
+			};
+		case "GHE_ERROR_CONNECTION_TIMED_OUT":
+			return {
+				title: "Connection timed out",
+				message: `We couldn't connect to <b>${url}</b>. Please make sure Github for Jira can connect to it and try again. <a href="https://support.atlassian.com/jira-cloud-administration/docs/integrate-with-github/#How-the-GitHub-for-Jira-app-fetches-data" target="_blank">Learn more</a>`
```
Comment (rachellerathbone) [inline static/js/jira-server-url.js:72]: Nit: Github -> GitHub.

File: src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts (hunk 2 above)
Comment (krazziekay) [reply src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts]: renamed to `TIMEOUT_PERIOD_MS`

File: static/js/jira-server-url.js (hunk 4 above)
Comment (krazziekay) [reply static/js/jira-server-url.js]: updated

**Review by rachellerathbone — APPROVED** [review]

**Pull request merged**